src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/SelfcareApiServiceImpl.scala

```diff
@@ -66,8 +65,8 @@ final case class SelfcareApiServiceImpl(
     val result: Future[Seq[SelfcareInstitution]] =
       for {
         userUuid        <- getUidFutureUUID(contexts)
-        institutions    <- selfcareV2ClientService.getInstitutions(userId = userUuid).map(_.map(_.toApi))
-        institutionsApi <- institutions.traverse(_.toFuture)
+        institutions    <- selfcareV2ClientService.getInstitutions(userId = userUuid)
+        institutionsApi <- Future.traverse(institutions)(_.toApi.toFuture)
```

Comment:
same, avoid leveraging on threads here
```suggestion
        institutionsApi <- institutions.traverse(_.toApi).toFuture
```